app/gui/src/project-view/util/ast/abstract.ts

```diff
@@ -202,32 +203,55 @@ export function substituteIdentifier(
   }
 }
 
+export function substituteQualifiedName(
+  expr: MutableAst,
+  pattern: QualifiedName | IdentifierOrOperatorIdentifier,
+  to: QualifiedName,
+): Ast
+export function substituteQualifiedName(
```

Comment:
I think it would be simpler to split this into two functions--the `pattern/to` version could be implemented on top of the new `substitution` version.